core/common/src/Sink.kt

```diff
@@ -21,295 +21,139 @@
 package kotlinx.io
 
 /**
- * A sink that keeps a buffer internally so that callers can do small writes without a performance
- * penalty.
+ * A sink that facilitates typed data writes and keeps a buffer internally so that caller can write some data without
```

Comment:
Here and in `Source` (and `Raw*` counterpart), it is probably worth explaining the following:

* Exception contract. Right now, it's a bit unclear what kind of exceptions are allowed from `read*` methods, it would be nice to streamline it a bit (probably it's a good idea to allow only `IOException` from the underlying sources and ISE/IAE for incorrect arguments/states). [An example](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization/-k-serializer/)
* Naming scheme for methods (https://github.com/Kotlin/kotlinx-io/issues/137#issuecomment-1594324577) explicitly spelled out is a great thing as well

Reply:
I'll wait a few days to make sure there will be no suggestions to correct the latest version of the naming scheme from #137 and then fix the docs.